### ISSUE TITLE
Applock does not always restrict access to app

### ISSUE THREAD

**Issue by loblawbob** [keepassium/KeePassium#72]
**Describe the bug**
I have Applock enabled and set to immediately, closing database set to never.
Applock does not reliably ask for TouchID after closing the app, instead going straight to the open database. The time between closing and reopening can be right away or a couple of hours, it can happen in both instances.
The bug does not happen two times in a row. If TouchID was skipped when opening, on the next time opening it will always ask for Touch ID.
If the app is force closed from the app switcher this bug does not occur.

**To Reproduce**
1. Applock enabled and set to immediately
2. Close database set to never and remember master key enabled
3. Have database open and go to home screen or a different app
4. Re-open Keepassium
5. Keepassium opens and enters the database without asking for TouchID

**Expected behavior**
With the settings set as described above Keepassium should always ask for authentication when opening the app without fail.

**Screenshots**

**User Information (please complete the following information):**
 - Device: iPhone 6s Plus
 - OS: iOS 13.2
 - App Version: v1.06.42

**Additional context**

**Comment by keepassium** [comment]
Thanks! Do you remember whether this happened on 13.1.3 as well?

(I am away from my dev machine at the moment, will investigate early next week)

**Comment by loblawbob** [comment]
On pre 13.2 I was unfortunately unable to open databases due to issue #71.

**Comment by keepassium** [comment]
Ah, right... So far this seems the only report of this kind, so hopefully the issue is not widespread. I will look into it in a couple days and get back to you.

**Comment by keepassium** [comment]
Yep, I can reproduce this on release iOS 13.2 (not on earlier versions, though). The cause is unclear yet, I am looking into it.

**Comment by loblawbob** [comment]
I don‘t know if this makes any difference but it is still happening on 13.2.2.

**Comment by keepassium** [comment]
Found and fixed it! 

Before 13.2, the system first told the app the results of the biometric auth, then notified that the app became active. On 13.2, the system first activates the app and then tells it the results of the biometric auth. The latter makes more sense, but did not play well with my state machine.

Hopefully, the fix did not break anything on other iOS versions. I've tested the fix on 13.1.2 and 13.2 devices (with Touch ID) and 13.1 and 12.2 simulators (with Face ID). 

I will submit the update to TestFlight later today. Apple will take a few days to approve it for beta testing, then if everything works properly the fix will be in the AppStore in another couple of days.

Thanks for reporting!
